akka-http/src/main/scala/akka/http/javadsl/server/directives/SecurityDirectives.scala

```diff
@@ -174,7 +174,7 @@ abstract class SecurityDirectives extends SchemeDirectives {
   def authenticateOAuth2AsyncOptional[T](realm: String, authenticator: JFunction[Optional[ProvidedCredentials], CompletionStage[Optional[T]]],
                                          inner: JFunction[Optional[T], Route]): Route = RouteAdapter {
     D.extractExecutionContext { implicit ctx =>
-      D.authenticateBasicAsync(realm, c => authenticator.apply(toJava(c)).toScala.map(_.asScala)).optional { t =>
+      D.authenticateOAuth2Async(realm, c => authenticator.apply(toJava(c)).toScala.map(_.asScala)).optional { t =>
```

Comment:
Let's pretend I never asked.

Nah, seriously though lets check it tomorrow, we try to avoid implicits for such things.